pkg/cmd/kubeconfig/kubeconfig_test.go

```diff
@@ -9,106 +9,137 @@ package kubeconfig_test
 
 	"github.com/golang/mock/gomock"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/tools/clientcmd"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 	"k8s.io/utils/pointer"
 
+	"github.com/gardener/gardenctl-v2/internal/util"
 	utilmocks "github.com/gardener/gardenctl-v2/internal/util/mocks"
 	cmdkubeconfig "github.com/gardener/gardenctl-v2/pkg/cmd/kubeconfig"
 	"github.com/gardener/gardenctl-v2/pkg/target"
 	targetmocks "github.com/gardener/gardenctl-v2/pkg/target/mocks"
 )
 
-var _ = Describe("Target Commands - Options", func() {
-	Describe("having an Options instance", func() {
+var _ = Describe("Kubeconfig Command - Options", func() {
+	var (
+		ctrl    *gomock.Controller
+		factory *utilmocks.MockFactory
+		manager *targetmocks.MockManager
+		options *cmdkubeconfig.TestOptions
+		t       target.Target
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		factory = utilmocks.NewMockFactory(ctrl)
+		manager = targetmocks.NewMockManager(ctrl)
+		options = cmdkubeconfig.NewOptions()
+		t = target.NewTarget("test", "project", "seed", "shoot")
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	Describe("Having a Kubeconfig command instance", func() {
 		var (
-			ctrl    *gomock.Controller
-			factory *utilmocks.MockFactory
-			manager *targetmocks.MockManager
-			options *cmdkubeconfig.TestOptions
-			t       target.Target
+			streams util.IOStreams
+			cmd     *cobra.Command
+			out     *util.SafeBytesBuffer
+			ctx     context.Context
+			config  clientcmd.ClientConfig
 		)
 
 		BeforeEach(func() {
-			ctrl = gomock.NewController(GinkgoT())
-			factory = utilmocks.NewMockFactory(ctrl)
-			manager = targetmocks.NewMockManager(ctrl)
-			options = cmdkubeconfig.NewOptions()
-			t = target.NewTarget("test", "project", "seed", "shoot")
+			streams, _, out, _ = util.NewTestIOStreams()
+			cmd = cmdkubeconfig.NewCmdKubeconfig(factory, streams)
+
+			ctx = context.Background()
+			config = &clientcmd.DirectClientConfig{}
+
+			factory.EXPECT().Manager().Return(manager, nil).AnyTimes()
+			factory.EXPECT().Context().Return(ctx)
+
+			manager.EXPECT().CurrentTarget().Return(t, nil)
+			manager.EXPECT().ClientConfig(ctx, t).Return(config, nil)
+		})
+
+		It("should execute the kubeconfig subcommand", func() {
+			cmd = cmdkubeconfig.NewCmdKubeconfig(factory, streams)
+			cmd.SetArgs([]string{"--output", "yaml"})
+			Expect(cmd.Execute()).To(Succeed())
+			Expect(out.String()).To(Equal(`apiVersion: v1
+clusters: null
+contexts: null
+current-context: ""
+kind: Config
+preferences: {}
+users: null
+`))
 		})
+	})
 
-		AfterEach(func() {
-			ctrl.Finish()
+	Describe("having an Options instance", func() {
+		var (
+			err       error
+			ctx       context.Context
+			config    clientcmd.ClientConfig
+			rawConfig clientcmdapi.Config
+		)
+
+		BeforeEach(func() {
+			ctx = context.Background()
+			config = &clientcmd.DirectClientConfig{}
```

Comment:
Do we need the DirectClientConfig. Why not directly create the clientcmdapi.Config